suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,250 @@
+        });
+    }, [hideAlert, showAlert, translate]);
+
+    const showAnotherEmptyAccountAlert = useCallback(() => {
+        showAlert({
+            title: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.title'),
+            description: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionPrimary',
+            ),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+            secondaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondary',
+            ),
+            onPressSecondaryButton: () => {
+                openLink(
+                    translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondaryUrl'),
+                );
+                hideAlert();
+            },
+        });
+    }, [hideAlert, showAlert, translate, openLink]);
+
+    const addCoinAddress = useCallback(
+        ({ network, type }: { network: Network; type?: AccountType }) => {
+            if (!device?.state) {
+                return false;
+            }
+
+            const accountType =
+                type ?? getAvailableAccountTypesForNetwork({ network })?.[0] ?? 'normal';
+
+            const currentAccountTypeAccounts = accounts.filter(
+                account => account.symbol === network.symbol && account.accountType === accountType,
+            );
+
+            // Do not allow adding more than 10 accounts of the same type
+            if (currentAccountTypeAccounts.length > 10) {
+                showTooManyAccountsAlert();
+                return false;
+            }
+
+            // Do not allow adding another empty account if there is already one
+            const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
+
+            if (emptyAccounts.length > 0) {
+                showAnotherEmptyAccountAlert();
+                return false;
+            }
+
+            const addNetworkAccount = async (deviceState: string) => {
+                setIsAddingCoinAccount(true);
+                const result = await dispatch(
+                    addNetworkAccountThunk({
+                        network,
+                        accountType,
+                        deviceState,
+                    }),
+                ).unwrap();
+
+                setAddedAccount(result.account);
+                setIsAddingCoinAccount(false);
+            };
+
+            addNetworkAccount(device.state);
+        },
+        [
+            device?.state,
+            getAvailableAccountTypesForNetwork,
+            accounts,
+            dispatch,
+            showTooManyAccountsAlert,
+            showAnotherEmptyAccountAlert,
+            setIsAddingCoinAccount,
+        ],
+    );
+
+    useEffect(() => {
```

Comment:
IMO unnecessary useEffect and also the state `addedAccount`. The navigation function could be called directly in the `addCoinAddress` callback where is the state set.

Reply:
044d93041c